### PR TITLE
:lipstick:  Image input block redesign #639

### LINKED ISSUE #639: :lipstick:  Image Input Block Redesign: block redesign
## User stories

As a user, I want the "Image Input Block" component in the app to be redesigned to enhance its usability and visual appeal.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Technical Description
The current design is: 
image

It should be changed to:
image

The files can be found at `libs/features/convs-mgr/stories/blocks/library/image-input-block/`. 

The designs are [here](https://www.figma.com/file/ZGHzSstXOYPMFU1EiqjEHD/Untitled?type=design&node-id=1%3A4&mode=design&t=Hf7DxaUUMQ3BWy57-1)

### Acceptance Criteria

1. The block should look exactly like the designs.
2. Use of `rem`, `em` as opposed to `px`.
3. All changes should be isolated to the following  `libs/features/convs-mgr/stories/blocks/library/image-input-block/`.

### PR DESCRIPTION
**Pull request by levy-web**
# Description

Re designed the image input block. Changed the image icon and added a menu svg icon to toggle between showing menu items and hiding them

Fixes #639 

## Type of change

Please delete options that are not relevant.

- [X] Design change (fix or feature that would cause existing functionality to not work as expected)

# Screenshot (optional)

Screenshot from 2023-11-05 10-17-28
Screenshot from 2023-11-05 10-17-34

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.html [hunk 1]
```diff
@@ -2,15 +2,21 @@
...
+
                 <h4 class="block-title"> {{ blockTitle | transloco }}</h4>
             </div>
 
-            <div fxLayout="row" class="icon-block">
+            <div fxLayout="row" *ngIf="isMenuOpen"  class="icon-block">
                 <span (click)="editBlock()" class="icons"><i class="far fa-edit"></i></span>
                 <span (click)="copyblock(block)" class="icons"><i class="far fa-clone"></i></span>
                 <span (click)="deleteBlock()" class="icons"><i class="far fa-trash-alt"></i></span>
             </div>
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.html:15]: Sync the latest changes and then use the menu set in the codebase. But you can switch the icon to the one you have over hear

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/model/icons-and-titles.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export const iconsAndTitles: any = {
   },
   50: {
     title: 'PAGE-CONTENT.BLOCK.TITLES.IMAGE-INPUT',
-    icon: 'fas fa-image',
+    icon: '/assets/icons/ph_image-light.svg',
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/model/icons-and-titles.ts:84]: Pull the latest changes. There's a switch between SVGs and i tags. We will deprecate the i tags after all blocks switch to svgs

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/image-input-block.jsplumb.ts [hunk 3]
```diff
@@ -21,7 +21,16 @@ export function _ImageInputBlockDecoratePlumb(block: ImageInputBlock, comp: Comp
...
+  });
+
+  jsPlumb.addEndpoint(comp.location.nativeElement, {
+    // Whether the anchor is target (Other Block -> This Block)
+    target: true,
+    cssClass:"block_endpoint",
+    endpoint: "Dot",
+    anchor: [1, 0.80, 0 , 1],
     maxConnections: -1
   });
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/image-input-block.jsplumb.ts:35]: No need for this. We've decided to just keep the original default option. So just-style that

File: libs/features/convs-mgr/stories/editor/src/lib/components/grouped-blocks/grouped-blocks.component.html [hunk 4]
```diff
@@ -2,7 +2,8 @@
   <ng-container *ngFor="let block of groupedBlocks">
     <div (click)="addBlock(block.type)" draggable="true" (dragend)="onDragEnd(block.type)" class="block">
       <div fxLayout="column" fxLayoutAlign="center center" class="blocks-main">
-        <i class="{{ block.blockIcon }}" id="icons"></i>
+        <i *ngIf="block.message !== 'Image Input'" class="{{ block.blockIcon }}" id="icons"></i>
+        <img *ngIf="block.message === 'Image Input'" [src]="block.blockIcon" />
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/editor/src/lib/components/grouped-blocks/grouped-blocks.component.html:6]: Use the switch-case already in the codebase at the block component section.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

File: libs/features/convs-mgr/stories/editor/src/lib/components/blocks-library/blocks-library.component.ts [hunk 5]
```diff
@@ -55,7 +55,7 @@ export class BlocksLibraryComponent implements OnInit, OnDestroy {
     { id: 'jump-story-block', type: StoryBlockTypes.JumpBlock, message: 'Jump', blockIcon: this.getBlockIcon(StoryBlockTypes.JumpBlock), blockCategory: 'questions-block' } as JumpBlock,
     // { id: 'io-multiple-input-block', type: StoryBlockTypes.MultipleInput, message: 'MultipleInput', blockIcon:this.getBlockIcon(StoryBlockTypes.MultipleInput) } as MultipleInputMessageBlock,
     { id: 'fail-block', type: StoryBlockTypes.FailBlock, message: 'Fail', blockIcon:this.getBlockIcon(StoryBlockTypes.FailBlock), blockCategory: 'questions-block' } as FailBlock,
-    { id: 'io-image-input-block' , type: StoryBlockTypes.ImageInput, message: 'Image Input', blockIcon:this.getBlockIcon(StoryBlockTypes.ImageInput), blockCategory: 'images-block' } as ImageInputBlock,
+    { id: 'io-image-input-block' , type: StoryBlockTypes.ImageInput, message: 'Image Input', blockIcon:this.getBlockIcon(StoryBlockTypes.ImageInput), svgIcon:this.getSvgIcon(StoryBlockTypes.ImageInput), blockCategory: 'images-block' } as ImageInputBlock,
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/editor/src/lib/components/blocks-library/blocks-library.component.ts:58]: leave this as blockIcon

File: libs/model/convs-mgr/stories/blocks/main/src/lib/story-block.interface.ts [hunk 6]
```diff
@@ -37,6 +37,9 @@ export interface StoryBlock extends IObject
   /** icon of the block on the editor */
   blockIcon: string;
 
+  /** svg icon of the block on the editor */
+  svgIcon?: string;
```
Comment (LemmyMwaura) [inline libs/model/convs-mgr/stories/blocks/main/src/lib/story-block.interface.ts:41]: No need for this variable

File: libs/features/convs-mgr/stories/editor/src/lib/components/grouped-blocks/grouped-blocks.component.html [hunk 7]
```diff
@@ -2,7 +2,12 @@
   <ng-container *ngFor="let block of groupedBlocks">
     <div (click)="addBlock(block.type)" draggable="true" (dragend)="onDragEnd(block.type)" class="block">
       <div fxLayout="column" fxLayoutAlign="center center" class="blocks-main">
-        <i class="{{ block.blockIcon }}" id="icons"></i>
+        <ng-container *ngIf="block.svgIcon; else showIcon">
+          <img src="{{ block.svgIcon }}" class="svgIcon">
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/editor/src/lib/components/grouped-blocks/grouped-blocks.component.html:6]: just use the blockIcon over here

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Comment by LemmyMwaura** [comment]
Kindly sync your changes and resolve the conflicts

**Comment by levy-web** [comment]
@LemmyMwaura , I have merged with develop and done with requested changes

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/utils/transformers/image-input-block.jsplumb.ts [hunk 8]
```diff
@@ -21,7 +21,7 @@ export function _ImageInputBlockDecoratePlumb(block: ImageInputBlock, comp: Comp
     target: true,
     cssClass:"block_endpoint",
     endpoint: "Dot",
-    anchor: "Left",
+    anchor: [0, 0.2 , 0, 1],
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/utils/transformers/image-input-block.jsplumb.ts:24]: Let's leave this as "Left" for now

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
One final change.

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/utils/transformers/image-input-block.jsplumb.ts (hunk 8 above)
Comment (levy-web) [reply libs/features/convs-mgr/stories/blocks/library/main/src/lib/utils/transformers/image-input-block.jsplumb.ts:24]: ok on it

**Comment by levy-web** [comment]
@LemmyMwaura , done with requested changes

**Review by LemmyMwaura — APPROVED** [review]

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**